bench/workgen/workgen.cxx

```diff
@@ -992,8 +993,22 @@ ThreadRunner::op_run(Operation *op)
         if (num_tables == 0)
             goto err;
 
-        tint = (random_value() % num_tables);
-        table_uri = _icontext->_dyn_table_names.at(tint);
+        // Select a random key.
+        auto it = _icontext->_dyn_tint.begin();
+        std::advance(it, random_value() % _icontext->_dyn_tint.size());
+        const std::string uri(it->first);
+
+        // Make sure the table is not marked for deletion.
+        bool marked_deleted =
+          std::find(_icontext->_dyn_tables_delete.begin(), _icontext->_dyn_tables_delete.end(),
+            uri) != _icontext->_dyn_tables_delete.end();
+        if (!marked_deleted) {
+            ++_icontext->_dyn_table_in_use[uri];
+            table_uri = uri;
+            tint = _icontext->_dyn_tint[uri];
+        } else {
+            goto err;
```

Comment:
Looks like if we select a table marked for deletion, the random table operation thread exits and doesn't try again to select a new table. We should address that but maybe in a new change request.